Model/CatalogIngestion/Command/PublishSourceItemsProductEvent.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2017-2022 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Model\CatalogIngestion\Command;
+
+use Bolt\Boltpay\Api\Data\ProductEventInterface;
+use Bolt\Boltpay\Api\ProductEventManagerInterface;
+use Bolt\Boltpay\Helper\Config;
+use Bolt\Boltpay\Logger\Logger;
+use Bolt\Boltpay\Model\Config\Source\Catalog\Ingestion\Events;
+use Magento\Catalog\Model\ProductFactory;
+use Magento\Inventory\Model\SourceItem;
+use Magento\Catalog\Model\ResourceModel\Product\Website\Link as ProductWebsiteLink;
+
+/**
+ * Source items based product event publisher
+ */
+class PublishSourceItemsProductEvent
+{
+    /**
+     * @var ProductEventManagerInterface
+     */
+    private $productEventManager;
+
+    /**
+     * @var RunInstantProductEvent
+     */
+    private $runInstantProductEvent;
+
+    /**
+     * @var Config
+     */
+    private $config;
+
+    /**
+     * @var ProductFactory
+     */
+    private $productFactory;
+
+    /**
+     * @var Logger
+     */
+    private $logger;
+
+    /**
+     * @var ProductWebsiteLink
+     */
+    private $productWebsiteLink;
+
+    /**
+     * @var array
+     */
+    private $instantProductEventUpdatedList = [];
+
+    /**
+     * @param ProductEventManagerInterface $productEventManager
+     * @param RunInstantProductEvent $runInstantProductEvent
+     * @param Config $config
+     * @param ProductFactory $productFactory
+     * @param ProductWebsiteLink $productWebsiteLink
+     * @param Logger $logger
+     */
+    public function __construct(
```

Comment:
Could we combine Model/CatalogIngestion/Command/PublishSourceItemsProductEvent.php and Model/CatalogIngestion/Command/RunInstantProductEvent.php into one file?

Reply:
Removed, all logic moved to 
Model/CatalogIngestion/ProductEventProcessor.php

https://github.com/BoltApp/bolt-magento2/pull/1560/files#diff-5034980d5cd6db4c4e94a8806840ed03c1ffb376c08b2c6c279e12e7dd77b3d2